source/Playnite/Controllers/GenericGameController.cs

```diff
@@ -133,5 +145,59 @@ private void Monitor_TreeDestroyed(object sender, EventArgs args)
             stopWatch.Stop();
             OnStopped(this, new GameControllerEventArgs(this, stopWatch.Elapsed.TotalSeconds));
         }
+
+        private void CheckGameImagePath()

```

Comment:
So these specific check methods should just return the path which is then assigned to the Game object in the Play method? I think i need to change the Game object because it is directly used to expand the variables as far as i understand it.

Reply:
You can create copy (for example via `GetClone` extension method) and modify that copy. I just don't like the idea of modify the input object since it's probably used elsewhere up the call stack.